libutils/string_lib.c

```diff
@@ -601,6 +602,120 @@ long StringToLongExitOnError(const char *str)
     return result;
 }
 
+/**
+ * @brief Converts a string of numerals in base 10 to a unsigned long 
+ *        integer.
+ *
+ * Result is stored in *value_out, return value should be checked.
+ * On error *value_out is unmodified and an error code is returned.
+ * Leading spaces in input string are skipped.
+ * String numeral must be terminated by NULL byte or space (isspace).
+ *
+ * @see StringToUlongExitOnError()
+ * @see StringToUlongDefaultOnError()
+ * @param[in] str String with numerals to convert, cannot be NULL
+ * @param[out] value_out Where to store result on success, cannot be NULL
+ * @return 0 on success, error code otherwise (see source code)
+ */
+int StringToUlong(const char *str, unsigned long *value_out)
+{
+    nt_static_assert(ERANGE != 0);
+    assert(str != NULL);
+    assert(value_out != NULL);
+
+    char *endptr = NULL;
+    unsigned long val;
+
+    errno = 0;
+    val = strtoul(str, &endptr, 10);
```

Comment:
```suggestion
    char *endptr = NULL;
    errno = 0;
    const unsigned long val = strtoul(str, &endptr, 10);
```

Declare and assign on the same line;

https://github.com/cfengine/core/blob/master/CONTRIBUTING.md#readability--maintainability

(You can make this change in StringToLong in a separate commit, if it's the same there).

Reply:
Good catch